core/src/main/java/org/apache/iceberg/LocationProviders.java

```diff
@@ -51,7 +52,7 @@ public static LocationProvider locationsFor(String location, Map<String, String>
             impl, LocationProvider.class), e);
       }
       try {
-        return ctor.newInstance(location, properties);
+        return ctor.newInstance(sanitizedLocation, properties);
```

Comment:
Nit: I generally prefer to rename the method argument so that there are fewer changes.